src/tags.rs

```diff
@@ -41,23 +43,25 @@ pub struct SharedTagsOptions {
 
 pub async fn tags<'a>(client: ApiClient<'a>, locale: &str, command: TagsSubcommand) {
     match command {
-        TagsSubcommand::ListAll { shared: SharedTagsOptions { long, filter } } => {
+        TagsSubcommand::ListAll {
+            shared: SharedTagsOptions { long, filter },
+        } => {
             let tags = client.get_all_tags().await.expect("valid tags");
-            list(
-                &tags,
-                locale,
-                filter.as_ref(),
-                long,
-            )
+            list(&tags, locale, filter.as_ref(), long)
         }
-        TagsSubcommand::List { shared: SharedTagsOptions { long, filter }, broadcaster } => {
-            let tags = client.get_stream_tags(broadcaster).await.expect("valid tags");
-            list(
-                &tags,
-                locale,
-                filter.as_ref(),
-                long,
-            )
+        TagsSubcommand::List {
+            shared: SharedTagsOptions { long, filter },
+            broadcaster,
+        } => {
+            let tags = client
+                .get_stream_tags(if broadcaster.is_some() {
+                    broadcaster.as_ref().unwrap()
+                } else {
+                    client.get_user()
+                })
```

Comment:
Maybe use a match here?
```rust
match broadcaster {
    Some(broadcaster) => broadcaster,
    None => client.get_user(),
}
```

Reply:
I tried, but kept fighting the borrow checker, will try again.